pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFormatSpec.java

```diff
@@ -47,28 +50,75 @@ public class DateTimeFormatSpec {
   public static final int MIN_FORMAT_TOKENS = 3;
   public static final int MAX_FORMAT_TOKENS = 4;
 
+  public static final int FORMAT_TIMEFORMAT_POSITION_PIPE = 0;
+  public static final int FORMAT_PATTERN_POSITION_PIPE = 1;
```

Comment:
Better to divide these static variables into 2 groups one for EPOCH format and other for SDF format. These looks a bit odd clumped together. 

You can also create class `EpochDateTimeFormat` and `SDFDateTimeFormat` that handle the processing part for this. It will also make it easier to add or change a specific format in the future without needing to change the variables

Reply:
I agree it looks vague. But I don't quite get my head around how this can be better achieved.

Does this look good?

```
  public static final int FORMAT_TIMEFORMAT_POSITION_PIPE = 0;
  
  public static final int DATETIME_PATTERN_POSITION = 1;
  public static final int DATETIME_TIMEZONE_POSITION = 2;
  
  public static final int EPOCH_UNIT_POSITION = 1;
  public static final int EOPCH_SIZE_POSITION = 2;
  
  public static final int MIN_FORMAT_TOKENS_PIPE = 1;
  public static final int MAX_FORMAT_TOKENS_PIPE = 3;
```

Reply:
Yes, this works for now. Just write a small comment mentionin that DATETIME ones are applicable for SDF patterns and EPOCH ones for well, EPOCH format

Reply:
Done